daemon/api.go

```diff
@@ -1082,16 +1085,37 @@ out:
 		origPath = form.Value["snap-path"][0]
 	}
 
-	info, err := readSnapInfo(tempPath)
-	if err != nil {
-		return InternalError("cannot read snap file: %v", err)
-	}
-	snapName := info.Name()
-
 	st := c.d.overlord.State()
 	st.Lock()
 	defer st.Unlock()
 
+	var snapName string
```

Comment:
I looked a bit into this, mostly because of constructing reasonable errors it wouldn't work that nicely without introducing some local error types which seems a bit overkill or having a strange signature